Service/Category/IndexBuilder.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service\Category;
+
+use Algolia\AlgoliaSearch\Exception\CategoryReindexingException;
+use Algolia\AlgoliaSearch\Exceptions\AlgoliaException;
+use Algolia\AlgoliaSearch\Helper\AlgoliaHelper;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Algolia\AlgoliaSearch\Helper\Entity\CategoryHelper;
+use Algolia\AlgoliaSearch\Logger\DiagnosticsLogger;
+use Algolia\AlgoliaSearch\Service\AbstractIndexBuilder;
+use Magento\Catalog\Model\ResourceModel\Category\Collection;
+use Magento\Framework\App\Config\ScopeCodeResolver;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\App\Emulation;
+
+class IndexBuilder extends AbstractIndexBuilder
+{
+    public function __construct(
+        protected ConfigHelper      $configHelper,
+        protected DiagnosticsLogger $logger,
+        protected Emulation         $emulation,
+        protected ScopeCodeResolver $scopeCodeResolver,
+        protected AlgoliaHelper     $algoliaHelper,
+        protected CategoryHelper    $categoryHelper
+    ){
+        parent::__construct($configHelper, $logger, $emulation, $scopeCodeResolver, $algoliaHelper);
+    }
+
+    /**
+     * @param int $storeId
+     * @param int $page
+     * @param int $pageSize
+     * @return void
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     * @throws \Exception
+     */
+    public function rebuildIndex(int $storeId, int $page, int $pageSize): void
```

Comment:
If we veer away from `rebuild` we could simplify this to `buildIndex` or just `build` (and include it in `Api\IndexBuilder`) ?

Reply:
I prefer `buildIndex` since it's more descriptive